include/cell.h

```diff
@@ -214,6 +214,66 @@ class Cell {
   //! Return rank
   unsigned rank() const;
 
+  //! Assign free surface
+  //! \param[in] free_surface boolean indicating free surface cell
+  void assign_free_surface(bool free_surface) { free_surface_ = free_surface; };
+
+  //! Return free surface bool
+  //! \retval free_surface_ indicating free surface cell
+  bool free_surface() { return free_surface_; };
```

Comment:
Could you make all unaltered functions as const and add inline when appropriate?